gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_lb_driver.py

```diff
@@ -0,0 +1,791 @@
+# Copyright 2013 New Dream Network, LLC (DreamHost)
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+from gbpservice.nfp.configurator.drivers.loadbalancer.v1.haproxy import (
+                                                    haproxy_rest_client)
+from gbpservice.nfp.configurator.lib import lb_constants
+from gbpservice.nfp.configurator.drivers.base import base_driver
+
+DRIVER_NAME = 'loadbalancer'
+PROTOCOL_MAP = {
+    lb_constants.PROTOCOL_TCP: 'tcp',
+    lb_constants.PROTOCOL_HTTP: 'http',
+    lb_constants.PROTOCOL_HTTPS: 'https',
+}
+BALANCE_MAP = {
+    lb_constants.LB_METHOD_ROUND_ROBIN: 'roundrobin',
+    lb_constants.LB_METHOD_LEAST_CONNECTIONS: 'leastconn',
+    lb_constants.LB_METHOD_SOURCE_IP: 'source'
+}
+REQUEST_RETRIES = 0
+REQUEST_TIMEOUT = 10
+
+
+LOG = logging.getLogger(__name__)
+
+
+class HaproxyOnVmDriver(base_driver.BaseDriver):
+    service_type = 'loadbalancer'
+    pool_to_device = {}
+
+    def __init__(self, plugin_rpc=None):
+        self.plugin_rpc = plugin_rpc
+
+    def _get_rest_client(self, ip_addr):
+        client = haproxy_rest_client.HttpRequests(
+                            ip_addr, lb_constants.HAPROXY_AGENT_LISTEN_PORT,
+                            REQUEST_RETRIES, REQUEST_TIMEOUT)
+        return client
+
+    def _get_device_for_pool(self, pool_id, context):
+        device = HaproxyOnVmDriver.pool_to_device.get(pool_id, None)
```

Comment:
All functions should have function headers. We can do this in a different PR though.